cylc/flow/task_job_mgr.py

```diff
@@ -667,6 +666,8 @@ def _poll_task_job_callback(self, suite, itask, cmd_ctx, line):
         finally:
             log_task_job_activity(ctx, suite, itask.point, itask.tdef.name)
 
+        jp_ctx = JobPollContext(job_log_dir, **items)
```

Comment:
@hjoliver out of the changes in this PR, this is the only one that looks like a possible reason for the hanging workflow... maybe prior to this change it was raising an error captured somewhere else? Or maybe this is triggering some other error? The other changes are really simple, and shouldn't cause this kind of issue I guess.

Reply:
Yeah I was wondering about that too.

Reply:
Maybe that line needs to be inside the try block.